File: TravelBook/app/src/main/java/com/ozgortech/travelbook/MainActivity.java
package com.ozgortech.travelbook;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.room.Room;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import com.ozgortech.travelbook.databinding.ActivityMainBinding;

import java.util.List;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.core.Scheduler;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    CompositeDisposable compositeDisposable; // kullan-at torba
    PlaceDatabase database;
    PlaceDao placeDao;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());

        compositeDisposable = new CompositeDisposable();
        database = Room.databaseBuilder(getApplicationContext(),PlaceDatabase.class,"Places")
                //.allowMainThreadQueries()
                .build();
        placeDao = database.placeDao();

        compositeDisposable.add(placeDao.getAll()
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(MainActivity.this::handleResponse)); // son satırda bize döndürlen listeyi nerede
        // ele alacağımızı belirtiyoruz.
    }

    private void handleResponse(List<Place> placeList){

        binding.recyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));
        PlaceAdapter adapter = new PlaceAdapter(placeList);
        binding.recyclerView.setAdapter(adapter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.travel_menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.add_place) {
            Intent intent = new Intent(MainActivity.this,MapsActivity.class);
            intent.putExtra("info","new");
            startActivity(intent);

        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear(); // torbayı temizle
    }
}

File: ArtBookKotlin/app/src/main/java/com/ozgortech/artbookkotlin/MainActivity.kt
package com.ozgortech.artbookkotlin

import android.content.Intent
import android.database.sqlite.SQLiteDatabase
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import androidx.recyclerview.widget.LinearLayoutManager
import com.ozgortech.artbookkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    lateinit var database: SQLiteDatabase
    lateinit var artArrayList: ArrayList<Art>
    lateinit var adapter: ArtAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        artArrayList = ArrayList()
        binding.recyclerView.layoutManager = LinearLayoutManager(this)
        adapter = ArtAdapter(artArrayList)
        binding.recyclerView.adapter = adapter
        database = this.openOrCreateDatabase("Arts", MODE_PRIVATE,null)
        database.execSQL("CREATE TABLE IF NOT EXISTS arts('artName' VARCHAR, 'artistName' VARCHAR, 'artYear' VARCHAR, 'image' VARCHAR)")

        var cursor = database.rawQuery("SELECT * FROM arts",null)

        var artNameIx = cursor.getColumnIndex("artName")
        var artistNameIx = cursor.getColumnIndex("artistName")
        var artYearIx = cursor.getColumnIndex("artYear")
        var imageIx = cursor.getColumnIndex("image")

        while (cursor.moveToNext()){
            var artName = cursor.getString(artNameIx)
            var artistName = cursor.getString(artistNameIx)
            var artYear = cursor.getString(artYearIx)
            var image = cursor.getString(imageIx)

            var art = Art(artName,artistName,artYear,image)
            artArrayList.add(art)
            adapter.notifyDataSetChanged()
        }

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        var menuInflater = getMenuInflater()
        menuInflater.inflate(R.menu.art_menu,menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.addArtItem){
            var intent = Intent(this,ArtActivity::class.java)
            intent.putExtra("info","new")
            startActivity(intent)
        }
        return super.onOptionsItemSelected(item)
    }
}